src/main/java/view/filter_panels/MovieFilter.java

```diff
@@ -0,0 +1,92 @@
+package view.filter_panels;
+
+import java.awt.Component;
+
+import javax.swing.BoxLayout;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import javax.swing.event.DocumentEvent;
+import javax.swing.event.DocumentListener;
+
+import interface_adapter.filter_list.FilterState;
+import interface_adapter.filter_list.FilterViewModel;
+
+/**
+ * Panel for filtering movies.
+ */
+public class MovieFilter extends JPanel implements Filter {
+
+    private static final String KEYWORDS = "keywords";
+    private static final String GENRES = "genres";
+    private static final String ACTORS = "actors";
+    // Text fields for the filter criteria. Set to empty by default to avoid null values.
+    private final JTextField keywordField = new JTextField("");
+    private final JTextField genreField = new JTextField("");
+    private final JTextField actorField = new JTextField("");
+    private final FilterViewModel filterViewModel;
+
+    public MovieFilter(FilterViewModel filterViewModel) {
+
+        this.filterViewModel = filterViewModel;
+        setLayout(new BoxLayout(this, BoxLayout.Y_AXIS));
+        createPanelHelper("Keywords", keywordField);
+        createPanelHelper("Genres", genreField);
+        createPanelHelper("Actors", actorField);
+
+        addListener(KEYWORDS, keywordField);
+        addListener(GENRES, genreField);
+        addListener(ACTORS, actorField);
+    }
+
+    private void createPanelHelper(String labelText, Component component) {
```

Comment:
Oh, feel free to resolve this btw, it's more of a comment than a change request.

Reply:
Yeah it would have to be through a code review (looking at dependencies); hopefully, we named the methods similarly enough that it's relatively easy to figure out if they share a purpose.